SofaKernel/framework/sofa/core/behavior/BaseMechanicalState.h

```diff
@@ -226,6 +226,9 @@ class SOFA_CORE_API BaseMechanicalState : public virtual BaseState
     /// build the jacobian of the constraint in a baseMatrix
     virtual void getConstraintJacobian(const ExecParams* params, sofa::defaulttype::BaseMatrix* J,unsigned int & off) = 0;
 
+    /// fill the jacobian matrix (of the constraints) with identity blocks on the provided list of nodes(dofs)
```

Comment:
Here the Identity is sparse and not necessary full (as you have a list of blocks). 
It is implemented in the Mechanical State because these Jacobians are stored in the Mechanical State. But this function can be called, of course from ForceFields / Constraints